2003/kangraemin/app/src/main/java/com/example/kangraemin/MainActivity.kt

```diff
@@ -50,43 +62,46 @@ class MainActivity : KangBaseActivity() {
             }
         compositeDisposable.add(whenSearchTextChanged)
 
-        val whenSearchButtonClicked = btn_search.clicks()
-            .map {
-                when (NetworkUtil().getConnectivityStatus(context = this)) {
-                    NetworkUtil.NetworkStatus.NOT_CONNECTED -> false
-                    else -> true
-                }
-            }
-            .subscribe { connectedToInternet ->
-                if (connectedToInternet) {
-                    rv_search_result.visibility = View.VISIBLE
-                    tv_network_error.visibility = View.GONE
-                    service.getSearchItems(
-                        display = "10",
-                        start = "1",
-                        query = et_search.text.toString()
-                    )
-                        .subscribeOn(Schedulers.io())
-                        .observeOn(AndroidSchedulers.mainThread())
-                        .subscribe({ responseMovieSearch ->
-                            adapter.data = responseMovieSearch.items
-                            adapter.notifyDataSetChanged()
-                        }, { it.printStackTrace() })
-                } else {
-                    rv_search_result.visibility = View.GONE
-                    tv_network_error.visibility = View.VISIBLE
-                }
-            }
-        compositeDisposable.add(whenSearchButtonClicked)
-
         val whenLogOutClicked = btn_logout.clicks()
             .subscribe {
-                editor.remove(LoginActivity.TAG_AUTO_LOGIN)
-                editor.apply()
-                startActivity(Intent(this, LoginActivity::class.java))
-                finish()
+                val deleteAuth = authRepo
+                    .deleteAuth(context = this)
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe({
+                        startActivity(Intent(this, LoginActivity::class.java))
+                        finish()
+                    }, { it.printStackTrace() })
+                compositeDisposable.add(deleteAuth)
             }
         compositeDisposable.add(whenLogOutClicked)
 
+        val whenArriveSearchResult = btn_search.clicks()
+            .toFlowable(BackpressureStrategy.BUFFER)
+            .doOnNext {
+                when (NetworkUtil().getConnectivityStatus(context = this)) {
+                    NetworkUtil.NetworkStatus.NOT_CONNECTED -> {
+                        rv_search_result.visibility = View.GONE
+                        tv_network_error.visibility = View.VISIBLE
+                    }
+                    else -> {
+                        rv_search_result.visibility = View.VISIBLE
+                        tv_network_error.visibility = View.GONE
+                    }
+                }
+            }
+            .map { et_search.text.toString() }
+            .switchMap { search(it) }
+            .startWith(search(""))
```

Comment:
startWith를 switchMap뒤에 놓으면 switchMap을 쓰는 의미가 없어요.
```suggestion
            .map { et_search.text.toString() }
             .startWith("")
            .switchMap { search(it) }
```

Reply:
수정하였습니다
7b806d9